### ISSUE TITLE
Is it intentional that type hints are ignored when using setattr?

### ISSUE THREAD

**Issue by phi-friday** [astral-sh/ty#2269]
### Question

```python
class A:
    a: int
    def __setattr__(self, key, value):
        ...

a = A()
a.a = "test"
```
https://play.ty.dev/fd8418ae-991f-43ee-b9f0-2f513d77efdb

In the code above, a is an int, so I expected an error to be raised when assigning "test", but it didn't occur.
It seems to be due to setatt, is this intentional or a bug?

### Version

playground b925ae506

**Comment by phi-friday** [comment]
Is this related to issue #1460?

**Comment by MichaReiser** [comment]
Yes, I think you're right. This is https://github.com/astral-sh/ty/issues/1460#issuecomment-3474720694. Adding type annotations to value in `__setattr__` makes ty raise an error on the assignment.
